### PR TITLE
SampleS coverage and competeness

### LINKED ISSUE #100: Coverage for all samples of a case
Add one more parameter to the query so that user can specify a list of samples

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Fix #100 

### How to test:
- Feed the coverage endpoints with a list of samples (could be a list with one element) and run a query for coverage over some genes/transcripts/exons 

### Expected outcome:
- The coverage and completeness fields should be lists instead of single values

### Review:
- [ ] Code approved by
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
- /coverage/d4/interval_file endpoint works 🆗 

image


<br><br>
- /coverage/samples/genes_coverage works as expected as well: 🆗 

image

```
[
  {
    "chromosome": "7",
    "completeness": [
      [
        "test_sample",
        [
          [
            20,
            0.8901363420913222
          ],
          [
            30,
            0.36031673745550524
          ]
        ]
      ]
    ],
    "ensembl_gene_id": "ENSG00000128513",
    "end": 124570037,
    "hgnc_id": 17284,
    "hgnc_symbol": "POT1",
    "interval_id": null,
    "mean_coverage": [
      [
        "test_sample",
        28.25178211288419
      ]
    ],
    "start": 124462440
  }
]
```

**Comment by northwestwitch** [comment]
Endpoint `/coverage/samples/transcripts_coverage` works as well: 🆗 

image

```
[
  {
    "chromosome": "7",
    "completeness": [
      [
        "test_sample",
        [
          [
            20,
            0.8889579958342976
          ],
          [
            30,
            0.35653967936714986
          ]
        ]
      ]
    ],
    "ensembl_gene_id": "ENSG00000128513",
    "end": 124570037,
    "hgnc_id": 17284,
    "hgnc_symbol": "POT1",
    "interval_id": null,
    "mean_coverage": [
      [
        "test_sample",
        28.279292186937408
      ]
    ],
    "start": 124462440
  }
]
```

<br><br>

Same for `/coverage/samples/exons_coverage` 🆗 

image


```
[
  {
    "chromosome": "7",
    "completeness": [
      [
        "test_sample",
        [
          [
            20,
            0.8557302570041424
          ],
          [
            30,
            0.28217005924012295
          ]
        ]
      ]
    ],
    "ensembl_gene_id": "ENSG00000128513",
    "end": 124570037,
    "hgnc_id": 17284,
    "hgnc_symbol": "POT1",
    "interval_id": null,
    "mean_coverage": [
      [
        "test_sample",
        269.20873768354465
      ]
    ],
    "start": 124462440
  }
]
```

**Comment by northwestwitch** [comment]
I am satisfied with how this works now, Ready for review!

File: src/chanjo2/crud/samples.py [hunk 1]
```diff
@@ -32,6 +29,12 @@ def get_samples(db: Session, limit: int = 100) -> List[SQLSample]:
     return db.query(SQLSample).limit(limit).all()
 
 
+def get_samples_by_name(db: Session, sample_names: List) -> List[SQLSample]:
```
Comment (henrikstranneheim) [inline src/chanjo2/crud/samples.py:32]:
```suggestion
def get_samples_by_name(db: Session, sample_names: List[str]) -> List[SQLSample]:
```

File: src/chanjo2/endpoints/coverage.py [hunk 2]
```diff
@@ -89,33 +92,43 @@ def d4_intervals_coverage(coverage_file_path: str, bed_file: bytes = File(...)):
...
-    try:
-        d4_file: D4File = set_d4_file(coverage_file_path=sample.coverage_file_path)
-    except Exception:
+def get_samples_coverage_file(
+    db: Session, samples: List[str]
+) -> Union[List[Tuple[str, D4File]], JSONResponse]:
+    """Return a list of sample names with relative D4 coverage files."""
+
+    samples_d4_files: List[Tuple[str, D4File]] = []
+    sqlsamples: List[SQLSample] = get_samples_by_name(db=db, sample_names=samples)
```
Comment (henrikstranneheim) [inline src/chanjo2/endpoints/coverage.py:101]:
```suggestion
    sql_samples: List[SQLSample] = get_samples_by_name(db=db, sample_names=samples)
```

File: src/chanjo2/endpoints/coverage.py [hunk 3]
```diff
@@ -89,33 +92,43 @@ def d4_intervals_coverage(coverage_file_path: str, bed_file: bytes = File(...)):
...
         )
-
-    return d4_file
-
-
-@router.post("/coverage/sample/genes_coverage", response_model=List[CoverageInterval])
-async def sample_genes_coverage(
+    for sqlsample in sqlsamples:
+        try:
+            d4_file: D4File = set_d4_file(
```
Comment (henrikstranneheim) [inline src/chanjo2/endpoints/coverage.py:110]:
```suggestion
            d4_file: D4File = get_d4_file(
```

**Review by henrikstranneheim — APPROVED** [review]
👍

**Pull request merged**